src/backend/distributed/commands/extension.c

```diff
@@ -491,6 +494,115 @@ PreprocessAlterExtensionUpdateStmt(Node *node, const char *queryString)
 }
 
 
+/*
+ * PostprocessAlterExtensionUpdateStmt is called after ALTER EXTENSION UPDATE
+ * command is executed by standard utility hook.
+ *
+ * Actually, we do not need such a post process function for ALTER EXTENSION
+ * UPDATE commands unless the extension is Citus itself. This is because we
+ * need to mark existing objects that are not included in distributed object
+ * infrastructure in older versions of Citus (but now should be) as distributed
+ * if we can update Citus successfully via standard utility hook.
+ */
+List *
+PostprocessAlterExtensionUpdateStmt(Node *node, const char *queryString)
+{
+	if (IsCreateAlterExtensionUpdateCitusStmt(node))
+	{
+		/* extension management can only be done via coordinator node */
+		EnsureCoordinator();
```

Comment:
This seems like it might cause ALTER EXTENSION to error out on workers.

We should just skip doing any work when !IsCoordinator(), since workers do not keep track of `citus.pg_dist_object`.